src/ProductConstructionService/ProductConstructionService.DependencyFlow/PullRequestBuilder.cs

```diff
@@ -199,10 +200,22 @@ public async Task<string> GenerateCodeFlowPRTitleAsync(
         return await CreateTitleWithRepositories($"[{targetBranch}] Source code changes from ", [update.SubscriptionId]);
     }
 
-    public async Task<string> GenerateCodeFlowPRDescriptionAsync(SubscriptionUpdateWorkItem update)
+    public async Task<string> GenerateCodeFlowPRDescriptionAsync(
+        SubscriptionUpdateWorkItem update,
+        string previousSourceCommit)
     {
-        var build = await _barClient.GetBuildAsync(update.BuildId)
-            ?? throw new Exception($"Failed to find build {update.BuildId} for subscription {update.SubscriptionId}");
+
+        var build = await _barClient.GetBuildAsync(update.BuildId);
+
+        string sourceDiffText;
+        if (previousSourceCommit != null && !string.IsNullOrEmpty(build.GitHubRepository))
```

Comment:
This must work for Azure DevOps repos too

Reply:
@adamzip FYI usually the author of the PR, when they address a comment (and they're sure it's addressed well), they mark it as resolved. If there's a follow-up question or you're unsure, you comment and leave it open.